### PR TITLE
Use CUDA_VISIBLE_DEVICES by default

### PR DESCRIPTION
**Pull request by benoit-cty**
Following https://github.com/mlco2/codecarbon/issues/567 it will be better for CodeCarbon to use the env variable CUDA_VISIBLE_DEVICES as a default value for the parameter gpu_ids.

I also drop Python 3.8 from CI as it is deprecated https://devguide.python.org/versions/ and use the same Python version everywhere in CI for consistency. I kept 3.8 in supported version to allow our users to continue to use CodeCarbon in 3.8 until it stop working.

### CODE REVIEW

File: codecarbon/external/hardware.py
Comment (benoit-cty) [inline codecarbon/external/hardware.py:107]: We stop raising an AssertionError that will break the user code. Instead, we just log a warning above.

File: tests/test_gpu.py
Comment (benoit-cty) [inline tests/test_gpu.py:225]: I need to do this because it seems the first calls to `nvmlDeviceGetTotalEnergyConsumption` are lost because of the new code of hardware.py:from_utils that made two calls to the class :
Previous code: `        return cls(gpu_ids=gpu_ids)`

New code:
```py
        gpus = cls(gpu_ids=gpu_ids)
        new_gpu_ids = gpus._get_gpu_ids()
        if len(new_gpu_ids) < gpus.num_gpus:
            logger.warning(
                f"You have {gpus.num_gpus} GPUs but we will monitor only {len(gpu_ids)} of them. Check your configuration."
            )
        return cls(gpu_ids=new_gpu_ids)
```

File: tests/test_emissions_tracker.py
Comment (benoit-cty) [inline tests/test_emissions_tracker.py:98]: I need to add one call because of the change in `from_utils` in `hardware.py`.

File: codecarbon/external/hardware.py
Comment (inimaz) [inline codecarbon/external/hardware.py:106]:
```suggestion
            if not set(self.gpu_ids).issubset(set(range(self.num_gpus))):
```

**Review by inimaz — CHANGES_REQUESTED** [review]
Thanks for this @benoit-cty !
Just small typo in one of the checks.

**Review by SaboniAmine — COMMENTED** [review]
LGTM, thanks Benoît! I'll let Inigo approve when his change request is done.

File: codecarbon/emissions_tracker.py
Comment (benoit-cty) [reply codecarbon/emissions_tracker.py:394]: If pynvml fail in `get_gpu_static_info()`, the result will be an empty list so `self._conf["gpu_count"]` will be set to 0.

File: codecarbon/external/hardware.py
Comment (benoit-cty) [reply codecarbon/external/hardware.py]: Nice catch !

**Review by inimaz — APPROVED** [review]

**Pull request merged**
